File: Leetcode/Algorithms/118. Pascal's Triangle.py
class Solution(object):
    def generate(self, numRows):
        L = [[1],[1,1]]
        if(numRows <= 2):
            return L[:numRows]
        for i in range(2,numRows):
            L.append([1])
            for j in range(1,i):
                L[i].append(L[i-1][j-1] + L[i-1][j])
            L[i].append(1)
        return L
            
class Solution(object):
    def generate(self, numRows):
        L = [[1]]
        for i in range(1, numRows):
            L.append(map(lambda x, y: x + y, [0] + L[i-1], L[i-1] + [0]))
        return L[:numRows]

File: Algorithms/Array/118. Pascal's Triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> pascal = new ArrayList<List<Integer>>();
        List<Integer> row = new ArrayList<Integer>();
        for(int i = 0; i < numRows; ++i) {
            row.add(0, 1);
            for(int j = 1; j < i; ++j) {
                row.set(j, row.get(j) + row.get(j + 1));
            }
            pascal.add(new ArrayList(row));
        }
        return pascal;
    }
}
